src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -614,8 +618,15 @@ private static void PopulatePackageFolders(IEnumerable<string> packageFolders, L
 
         internal static LockFileLibrary CreateLockFileLibrary(LocalPackageInfo package, string sha512, string path)
         {
+            var hasTools = package.Files.Any(static file => HasTools(file));
```

Comment:
The previous loop has doing work to avoid boxing, but this is undoing that. 
Is there some other allocation benefit that I'm missing?